symfony/plugins/orangehrmPimPlugin/Api/EmployeeSkillAPI.php

```diff
@@ -0,0 +1,355 @@
+use OrangeHRM\Pim\Dto\EmployeeSkillSearchFilterParams;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointCreateResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointDeleteResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointGetAllResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointGetOneResult;
+use OrangeHRM\Core\Api\V2\Serializer\EndpointUpdateResult;
+use OrangeHRM\Entity\EmployeeSkill;
+use OrangeHRM\Pim\Service\EmployeeSkillService;
+use OrangeHRM\Pim\Api\Model\EmployeeSkillModel;
+use OrangeHRM\Core\Api\V2\Exception\RecordNotFoundException;
+use OrangeHRM\Core\Exception\DaoException;
+use Exception;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Exception\ServiceException;
+
+class EmployeeSkillAPI extends Endpoint implements CrudEndpoint
+{
+    use EntityManagerHelperTrait;
+
+    public const PARAMETER_SKILL_ID = 'skillId';
+    public const PARAMETER_YEARS_OF_EXP = 'yearsOfExperience';
+    public const PARAMETER_COMMENTS = 'comments';
+
+    public const FILTER_YEARS_OF_EXP = 'yearsOfExperience';
+    public const FILTER_COMMENTS = 'comments';
+
+    public const PARAM_RULE_YEARS_OF_EXP_MAX_LENGTH = 2;
+    public const PARAM_RULE_COMMENTS_MAX_LENGTH = 100;
+
+    /**
+     * @var null|EmployeeSkillService
+     */
+    protected ?EmployeeSkillService $employeeSkillService = null;
+
+    /**
+     * @return EmployeeSkillService
+     */
+    public function getEmployeeSkillService(): EmployeeSkillService
+    {
+        if (is_null($this->employeeSkillService)) {
+            $this->employeeSkillService = new EmployeeSkillService();
+        }
+        return $this->employeeSkillService;
+    }
+
+    /**
+     * @param EmployeeSkillService $employeeSkillService
+     */
+    public function setEmployeeSkillService(EmployeeSkillService $employeeSkillService): void
+    {
+        $this->employeeSkillService = $employeeSkillService;
+    }
+
+    /**
+     * @return EndpointGetOneResult
+     * @throws RecordNotFoundException
+     * @throws Exception
+     */
+    public function getOne(): EndpointGetOneResult
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+        $skillId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $employeeSkill = $this->getEmployeeSkillService()->getEmployeeSkillDao()->getEmployeeSkillById(
+            $empNumber,
+            $skillId
+        );
+        if (!$employeeSkill instanceof EmployeeSkill) {
```

Comment:
no need if condition, just use `$this->throwRecordNotFoundExceptionIfNotExist($employeeSkill, EmployeeSkill::class);`